internal/catalogtask/catalogtask.go

```diff
@@ -0,0 +1,122 @@
+package catalogtask
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/common"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/logger"
+)
+
+type TaskOperator interface {
+	Get() (*common.RequestMessage, error)
+	Update(data map[string]interface{}) error
+}
+
+type CatalogTask struct {
+	url  string
+	ctx  context.Context
+	glog logger.Logger
+}
+
```

Comment:
@bzwei we need a Factory Interface here
```
type CatalogTaskFactory interface {
   Create(ctx context.Context, url string) TaskOperator
}

type DefaultFactory struct {
}

func (df DefaultFactory) Create(ctx context.Context, url string) *CatalogTask {
    glog := logger.GetLogger(ctx)
    return &CatalogTask{ctx: ctx, url: url, glog: glog}
}
```

In our tests we would have a TestFactory which can do something like
```
type MockCatalogTask struct {
...
}
type TestCatalogFactory struct {
}

func (tdf TestCatalogFactory) Create(ctx context.Context, url string) *MockCatalogTask {
    glog := logger.GetLogger(ctx)
    return &MockCatalogTask{ctx: ctx, url: url, glog: glog}
}
```

We can pass the CatalogTaskFactory to our methods in the request